### PR TITLE
Modify arguments of Axis.__init__ 

### PR DESCRIPTION
**Pull request by alixdamman**

### CODE REVIEW

File: larray/core.py [hunk 1]
```diff
@@ -825,27 +825,40 @@ class Axis(object):
...
     >>> sex
     Axis('sex', ['M', 'F'])
     """
     # ticks instead of labels?
     # XXX: make name and labels optional?
-    def __init__(self, name, labels):
+    def __init__(self, name_deprecated=None, labels=None, name=None):
+        if name_deprecated is not None:
+            warnings.warn("Building an Axis by passing its name first is deprecated. "
+                          "An Axis should build as \"gender=Axis(labels=['M','F'], name='gender')\"."
```
Comment (gdementen) [inline larray/core.py]: should be created using?

File: larray/core.py [hunk 2]
```diff
@@ -4036,9 +4049,7 @@ def set_axes(self, axes_to_replace=None, new_axis=None, inplace=False, **kwargs)
             return LArray(self.data, axes, title=self.title)
 
     def with_axes(self, axes):
-        warnings.warn("LArray.with_axes is deprecated, "
-                      "use LArray.replace_axes instead",
-                      DeprecationWarning)
+        warnings.warn("LArray.with_axes is deprecated, use LArray.replace_axes instead", DeprecationWarning)
```
Comment (gdementen) [inline larray/core.py]: this warning should be updated... replace_axes does not exist anymore :)

**Comment by gdementen** [comment]
I just got an idea of a potentially better upgrade path for users. We'll discuss this tomorrow.

**Comment by gdementen** [comment]
```Axis(labels, name_deprecated=None, name=None)```

File: larray/core.py [hunk 3]
```diff
@@ -876,38 +876,53 @@ class Axis(object):
...
+        if isinstance(labels, basestring):
+            if '=' in labels:
+                name, labels = [o.strip() for o in labels.split('=')]
+            elif '..' not in labels and ',' not in labels:
+                warnings.warn("Arguments 'name' and 'labels' have been inverted in version 0.22 of larray. "
+                              "Be aware that the passed argument '{}' will be used to build labels and "
+                              "not to set the name of the axis".format(labels))
+                _name = labels
+                labels = name
+                name = _name
```
Comment (gdementen) [inline larray/core.py]: in Python, this is usually done like this: `name, labels = labels, name`

Comment (alixdamman) [inline larray/core.py]: I have a long history with Fortran/C++/Java, sorry.

Comment (gdementen) [inline larray/core.py]: :)

**Comment by alixdamman** [comment]
I have updated the warning message in Axis.__init__. Is it explicit enough?

**Comment by gdementen** [comment]
now that you mention it. I should contain a reference to LArray(), and/or add the correct stacklevel to warnings.warn so that users see what line of their code they need to be change.

**Pull request merged**